integrate/romberg.go

```diff
@@ -0,0 +1,67 @@
+// Copyright ©2019 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package integrate
+
+import (
+	"math"
+	"math/bits"
+)
+
+// Romberg returns an approximate value of the integral
+//  \int_a^b f(x)dx
+// computed using the Romberg's method. The function f is given
+// as a slice of equally-spaced samples, that is,
+//  f[i] = f(a + i*dx)
+// and dx is the spacing between the samples.
+//
+// The length of f must be 2^k + 1, where k is a positive integer,
+// and dx must be positive.
+//
+// See https://en.wikipedia.org/wiki/Romberg%27s_method for a description of
+// the algorithm
+func Romberg(f []float64, dx float64) float64 {
+	if len(f) < 3 {
+		panic("integral: invalid slice length: must be at least 3")
+	}
+
+	if dx <= 0 {
+		panic("integral: invalid spacing: must be larger than 0")
+	}
+
+	n := len(f) - 1
+	k := bits.Len(uint(n - 1))
+
+	if len(f) != 1<<uint(k)+1 {
+		panic("integral: invalid slice length: must be 2^k + 1")
+	}
+
+	work := make([]float64, 2*k)
+	prev := work[:k]
+	curr := work[k:]
+
+	h := dx * float64(len(f)-1)
+	prev[0] = (f[0] + f[len(f)-1]) * 0.5 * h
+
+	var estimate float64
```

Comment:
No need to have this here, move it into the loop body.

Reply:
Now I see it's here because of my suggestion, sorry :-)

Reply:
I'll take set it back

Reply:
Ah, I remember. When I wrote to move this line just before the loop, I meant the inner `j` loop because that's the only place where it's used.